### DISCUSSION TITLE
Help with getting this to work

### DISCUSSION THREAD

**Discussion by StressedOut09** [typekcz/beepboop-steam#26] [General]
I have no clue what I`m doing, Id find it helpful if there was a video to go along with. Though any help would be great. Im unfamiliar with pretty much everything to do here. Could someone give a very indepth guide on what to do (for windows machine) from start to finish? (ive setup a steam account for it). Currently i get this:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/1105c26e-3a54-4808-a83f-22eaa9ac7795)

I`m so confused.

**Comment by typekcz** [comment]
Hi,
sorry that there's no in-depth tutorial, but you got pretty far on your own.
You basically got it running, just probably one thing is missing.

You need to put `config.json` into the project folder. Maybe you have put it there with wrong name like `config.json.txt`. Make sure you have [file extensions enabled](https://www.wikihow.tech/Show-File-Extensions-on-Windows). The content of `config.json` should look like this:

```json
{
	"baseUrl": "http://localhost:8080/",
	"steam": {
		"groupName": "Testy Mc Test Face",
		"channelName": "Voicey Mc Voice Face"
	}
}
```
Here you can change group and channel where the bot will connect.

Also make sure you have `.env` file (also be carefull to not have any the extension) where you put username and password:
```
STEAM_USERNAME=BeepBoop
STEAM_PASSWORD=sUp3rH4rdC0r3p@s5w0rd
```

Let me know if you have any more issues.

**Reply by StressedOut09** [reply]
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/581e45b6-b622-4dca-b7c7-6dadc95bef4f)
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/fd3f010f-756b-43ab-9a95-32067982057f)
For some reason it created a config.json folder rather than a file so i just put config in there
Just created the .env file with login

Still getting the following:
![Uploading image.png…]()

**Reply by typekcz** [reply]
Docker probably created it as a directory when it was missing.

You have to delete `config.json` folder and put the `config.json` file instead of it. Also I see that you have there `pass.env` file, but it should be just `.env` that's the full name of the file.

**Reply by StressedOut09** [reply]
Yeah deleted folder and replaced with config.json

Folder is:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/6be96ec2-93a0-4d52-aa23-28c6a0ba4fe1)

Config.json looks like:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/c4b0475c-9eea-4523-98b0-47de5d50f448)


Error in docker:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/23ae8ba3-8640-4bf7-b3b5-e971b84bbaa5)

**Reply by typekcz** [reply]
You need to recreate the container. Run this to delete the containers:
```
docker-compose -f docker-compose.web.yml down
```
and then create and start it with:
```
docker-compose -f docker-compose.web.yml up
```

**Reply by StressedOut09** [reply]
Thanks I`m Trying that now

**Reply by StressedOut09** [reply]
I Get this when trying to authenticate with steam
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/e7490831-20d5-4381-b051-6da7f22a442f)

**Reply by StressedOut09** [reply]
I got the Code but how do i Input it?

(Also thanks for being so quick with responses, Can`t wait to use this)

**Reply by typekcz** [reply]
Open `localhost:8080/` that's the main interface and there should be popup for Steam Guard code.
You can also check `localhost:8080/screen` to see what is happening.

**Reply by StressedOut09** [reply]
Sadly When checking the screen this is all that i see,
Also didn`t send through a code this time...
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/291d7ec2-5b27-4e26-a60e-9210d347b83e)
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/58d5dd65-68e5-415e-8dc5-150c29894727)

**Reply by StressedOut09** [reply]
I get this in the docker:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/0cc13be5-8e9d-4e7a-b438-5a8db5042b7e)
could this be why?

**Reply by typekcz** [reply]
Look like Steam changed something on their login page, so that the bot can't find login form inputs. I need to make an update to fix this.

**Reply by typekcz** [reply]
I release a new version. Please download it and try again.

**Reply by StressedOut09** [reply]
Still Can`t get it to work still showing this:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/b149e1e6-82b2-462e-8de0-ddd00757be86)

I may make a detailed youtube vid as well for anyone else trying to set this up

**Reply by typekcz** [reply]
Have you downloaded the new version rebuild the container with `docker-compose -f docker-compose.web.yml up --build`?
It should show version 2.1.3 in here:
![obrazek](https://github.com/typekcz/beepboop-steam/assets/1359184/703ec35d-88b5-4481-8756-4f224ffe698f)

**Reply by StressedOut09** [reply]
Unable to get it to play the songs, it just stays there quietly:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/b991c6ee-f0b8-4a37-bbe8-ca7005bd6599)

Checking docker i`m just getting a bunch of these:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/11be9696-9ba6-40e3-ba3b-24c186862a71)

Screen looks like this:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/4bd27a98-61d6-43cc-8460-6e144d33ea49)

**Reply by typekcz** [reply]
That should have worked.
Errors you sent are unrelated, that's just some garbage coming from Steam Chat itself.
After sending the chat command (or you can try in the web interface) there should be this in the log:
```
playUrl https://www.youtube.com/watch?v=xaKSGeRr3N0
youtube detected
```
after that can be some error, but that should be reported by bot back to the chat, so not sure what's going on.
I would suggest to try the web interface, chat commands are hooked through the notifications so maybe the account has disabled notifications?

**Reply by StressedOut09** [reply]
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/bad3295c-014f-4b5f-86f5-be2aafa6abfa)
Think it works now, looks alot different, restarted PC and it may be working

Nevermind:
Still no music coming through: And i downloaded the 2.1.3 file

![image](https://github.com/typekcz/beepboop-steam/assets/160611175/7b05a2cb-91cd-492d-a707-3845c6c39429)


Error:
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/fd34e0ff-2631-4325-9851-9d0cf93aa52f)

Playing from in local host comes up with following error: 
![image](https://github.com/typekcz/beepboop-steam/assets/160611175/9d179227-d1f8-4c67-816f-3936bae8e341)

**Reply by typekcz** [reply]
It could be that specific youtube video cannot be played for some reason. You would have to find relevant part in the log as I said before those page error are unrelated and the relevat part should be after `playUrl ...`.
I would try to play the crab rave, that is what I usually test on and should work, if not then there's some other weird issue.

**Reply by StressedOut09** [reply]
tried playing crab rave and it no work

**Reply by typekcz** [reply]
That's weird.
You can try to hunt down the error message as I described before or you can try modifying `Dockerfile.web` to look like this:
```
FROM zenika/alpine-chrome:with-node

ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD 1
ENV PUPPETEER_EXECUTABLE_PATH /usr/bin/chromium-browser
COPY --chown=chrome . /beepboop
WORKDIR /beepboop
RUN npm install
RUN npm install ytdl-core@latest
ENTRYPOINT ["tini", "--"]
CMD ["npm", "start"]
```

**Reply by MissionBox** [reply]
Try running CMD or Powershell in this file directory. Usually works for me when I run into this issue
